src/mon/MonClient.cc

```diff
@@ -1136,11 +1136,13 @@ void MonClient::handle_get_version_reply(MMonGetVersionReply* m)
 
 AuthAuthorizer* MonClient::build_authorizer(int service_id) const {
   Mutex::Locker l(monc_lock);
-  assert(auth || active_con->get_auth());
-  if (auth)
+  if (auth) {
     return auth->build_authorizer(service_id);
-  else
-    return active_con->get_auth()->build_authorizer(service_id);
```

Comment:
because the `active_con->get_auth()` will be promoted to `this->auth`, once the client is authorized. and it does not help to return a not-yet-authorized `CephxClientHandler`. i will update the commit message.